src/main/java/nz/cri/gns/NZSHM22/opensha/polygonise/NZSHM22_PolygonisedDistributedModelBuilder.java

```diff
@@ -241,4 +246,44 @@ public void apply(FaultSystemRupSet rupSet) {
         List<Double> mMins = gridMmins(newGrid, rupSet);
         rupSet.addModule(new NZSHM22_PolygonisedDistributedModel(oldGrid, newGrid, mMins));
     }
+
+    /**
+     * This reproduces the file posted at
+     * https://nshmrevisionproject.slack.com/archives/C02E5PXMB1D/p1661301108709039
+     */
+    public static void reproduceNZSHM22() throws IOException, DocumentException {
+        String outputPath = "/tmp";
+        ParameterRunner parameterRunner = new ParameterRunner(Parameters.NZSHM22.INVERSION_CRUSTAL);
+        NZSHM22_CrustalInversionRunner runner = NZSHM22_PythonGateway.getCrustalInversionRunner();
+        parameterRunner.ensurePaths();
+        parameterRunner.setUpCrustalInversionRunner(runner);
+
+        // set up odd config
+        runner.setSpatialSeisPDF(NZSHM22_SpatialSeisPDF.NZSHM22_1346);
+        runner.setMinMags(6.8, 6.5);
+
+        // set up polygoniser specific config
+        runner.setPolyBufferSize(runner.getPolyBufferSize(), 3);
+        runner.setPolygonizer(4, "LINEAR", 40);
+
+        // shortcut the inversion as much as possible
+        runner.setIterationCompletionCriteria(10);
+        runner.setSelectionIterations(1);
+        runner.setRepeatable(true);
+        runner.setInversionAveraging(false);
+
+        FaultSystemSolution solution = runner.runInversion();
+        Path solutionPath = Path.of(outputPath, "inversionSolution.zip");
+        solution.write(solutionPath.toFile());
+
+        ZipTools.copyFromZipFile(
+                solutionPath,
+                "ruptures/NZSHM22_PolygonisedDistributedModel.csv",
+                Path.of(outputPath, "NZSHM22_PolygonisedDistributedModel.csv"),
```

Comment:
this file isn't in the inversion solutions I've seen (e.g. downloaded from weka for the 2022 SRM). Is that because we don't normally call `setPolygonizer` when running an inversion?

Reply:
correct!